### DISCUSSION TITLE
Clang Requirement in cpp2py "magic"

### DISCUSSION THREAD

**Discussion by andrewkhardy** [TRIQS/triqs#906] [General]
In the TwoParticleResponse examples for 02s-Lindhard, the command 

`using g_k_w_vt = gf_view<prod<brzone, imfreq>, scalar_valued>;`

gives the error  `ModuleNotFoundError: No module named 'clang'`

However I have installed Clang via both apt-get and pip for python and confirm they are installed 

`andrewhardy@pop-os:~$ clang --version

Ubuntu clang version 14.0.0-1ubuntu1.1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
andrewhardy@pop-os:~$ clang++ --version

Ubuntu clang version 14.0.0-1ubuntu1.1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
`

Is there some sort of symlink I am missing or what do I need to do to get this running?

**Comment by the-hampel** [comment]
Hi @andrewkhardy,

the cpp2py magic is always a bit tricky to get working. We are currently reworking the cpp2py package to make things easier. 

Regarding your problem: How did you install triqs? Via the debian packages or manually installed? 

@Wentzell Does the cpp2py cell magic work with the packaged version of triqs or does triqs / cpp2py need to be compiled with clang?

**Reply by andrewkhardy** [reply]
I installed via the Ubuntu/Debian package.

**Comment by Wentzell** [comment]
Dear @andrewkhardy,

You are missing the `python3-clang` Ubuntu package.
Installing this should be enough to get the cpp2py magic to work.
